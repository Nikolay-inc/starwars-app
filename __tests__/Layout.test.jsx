import React from 'react';
import { render } from '@testing-library/react';
import RootLayout from '../src/app/layout';

const ChildComponent = () => <div data-testid="child">Child Component</div>;

describe('RootLayout Component', () => {
  test('renders header component', () => {
    const { getByTestId } = render(
      <RootLayout>
       <ChildComponent />
      </RootLayout>
    );
    const headerElement = getByTestId('header');
    expect(headerElement).toBeInTheDocument();
  });

  test('renders footer component', () => {
    const { getByTestId } = render(
      <RootLayout>
       <ChildComponent />
      </RootLayout>
    )
    const footerElement = getByTestId('footer');
    expect(footerElement).toBeInTheDocument();
  });

  test('renders children components', () => {
    const { getByTestId } = render(
      <RootLayout>
        <ChildComponent />
      </RootLayout>
    );
    const childElement = getByTestId('child');
    expect(childElement).toBeInTheDocument();
  });

  test('matches snapshot', () => {
    const { container } = render(
      <RootLayout>
       <ChildComponent />
      </RootLayout>
    )
    expect(container).toMatchSnapshot();
  });
});
