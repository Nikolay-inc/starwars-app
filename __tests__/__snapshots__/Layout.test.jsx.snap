// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RootLayout Component matches snapshot 1`] = `
<div>
  <html
    lang="en"
  >
    <body
      class="className"
    >
      <header
        class="header"
        data-testid="header"
      >
        StarWars Heroes
      </header>
      <div
        class="layout"
      >
        <div
          data-testid="child"
        >
          Child Component
        </div>
      </div>
      <footer
        class="footer"
        data-testid="footer"
      >
        <span>
          StarWars
        </span>
      </footer>
    </body>
  </html>
</div>
`;
